zio-http/src/main/scala/zhttp/service/Server.scala

```diff
@@ -120,10 +121,11 @@ object Server {
       reqHandler      = settings.app.compile(zExec, settings, ServerTimeGenerator.make)
       init            = ServerChannelInitializer(zExec, settings, reqHandler)
       serverBootstrap = new ServerBootstrap().channelFactory(channelFactory).group(eventLoopGroup)
-      _ <- ChannelFuture.asManaged(serverBootstrap.childHandler(init).bind(settings.address))
-
+      channel         = serverBootstrap.childHandler(init).bind(settings.address)
```

Comment:
This is an effectful computation. Keep it as before, ie pass by name.

Reply:
Won't the effect be run later on during the useForever call?

Reply:
It's not the same. The previous approach can be caught and is a checked exception.
Here however, it will be thrown as an unchecked exception and you will not be able to catch it.